install_dependencies.sh

```diff
@@ -13,189 +13,256 @@ usage()
     echo "[--validate, -v]            Validate that required packages are installed"
     echo "[--docker, -d]              Specialize execution for docker"
     echo "[--no-distributed]          Don't install distributed compute dependencies (OpenMPI)"
-    echo "[--mode, -m <mode>]         Select installation mode: runtime, build, baremetal"
     exit 1
 }
 
-FLAVOR=`grep '^ID=' /etc/os-release | awk -F= '{print $2}' | tr -d '"'`
-VERSION=`grep '^VERSION_ID=' /etc/os-release | awk -F= '{print $2}' | tr -d '"'`
-MAJOR=${VERSION%.*}
-ARCH=`uname -m`
-
-if [ $FLAVOR != "ubuntu" ]; then
-    echo "Error: Only Ubuntu is supported"
-    exit 1
-fi
-
-UBUNTU_CODENAME=$(grep '^VERSION_CODENAME=' /etc/os-release | awk -F= '{print $2}' | tr -d '"')
-export UBUNTU_CODENAME
+detect_os() {
+    if [ -f /etc/os-release ]; then
+        . /etc/os-release
+        OS_ID="$ID"
+        OS_VERSION="$VERSION_ID"
+        OS_CODENAME="${VERSION_CODENAME:-}"
```

Comment:
```suggestion
        OS_CODENAME="${UBUNTU_CODENAME:VERSION_CODENAME}"
```
This OS_CODENAME should use UBUNTU_CODENAME if it exists. This is because it is used for registering package repositories.
Ubuntu has both VERSION_CODENAME and UBUNTU_CODENAME set to the same value, so there is no issue.
For example, in Linux Mint 22.1, VERSION_CODENAME is the Mint code name “xia,” while UBUNTU_CODENAME is the base Ubuntu 24.04 code name “noble.”
In this case, package manager registration should be done using ‘noble’ rather than “xia,” so UBUNTU_CODENAME should be used.